sdk/core/Azure.Core/src/Shared/AuthorizationChallengeParser.cs

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net;
+
+#nullable enable
+
+namespace Azure.Core
+{
+    /// <summary>
+    /// A helper class for parsing Authorization challenge headers.
+    /// </summary>
+    internal static class AuthorizationChallengeParser
+    {
+        /// <summary>
+        /// Parses the specified parameter from a challenge hearder found in the specified <see cref="Response"/>.
+        /// </summary>
+        /// <param name="response">The <see cref="Response"/> to parse.</param>
+        /// <param name="challengeScheme">The challenge scheme containing the <paramref name="challengeParameter"/>. For example: "Bearer"</param>
+        /// <param name="challengeParameter">The parameter key name containing the value to return.</param>
+        /// <returns>The value of the parameter name specified in <paramref name="challengeParameter"/> if it is found in the specified <paramref name="challengeScheme"/>.</returns>
+        public static string? GetChallengeFromResponse(Response response, string challengeScheme, string challengeParameter)
```

Comment:
Should this name be `GetChallengeParameterFromResponse` since it returns a single parameter value not the whole challenge string?